x/finality/keeper/power_table_test.go

```diff
@@ -404,124 +433,323 @@ func FuzzVotingPowerTable_ActiveFinalityProviderRotation(f *testing.F) {
 			h.CreateCovenantSigs(r, covenantSKs, delMsg, del, 10)
 			h.AddInclusionProof(stakingTxHash, btcHeaderInfo, inclusionProof, 30)
 
+			fHooks.EXPECT().AfterBtcDelegationActivated(
+				gomock.Any(),
+				fp.Address(),
+				del.Address(),
+				false,
+				gomock.Any(), // this will depend on the active set after for loop
+				stakingValue,
+			).Times(1)
+			// TODO: finalize expected hooks values
+
 			// record voting power
-			fpsWithMeta = append(fpsWithMeta, &types.FinalityProviderWithMeta{
-				BtcPk:       fp.BtcPk,
-				VotingPower: stakingValue,
+			fpsWithMeta = append(fpsWithMeta, &FinalityProviderWithMetaCostaker{
+				FinalityProviderWithMeta: types.FinalityProviderWithMeta{
+					BtcPk:       fp.BtcPk,
+					VotingPower: stakingValue,
+				},
+				Addr:       fp.Address(),
+				PrevStatus: btcstktypes.FinalityProviderStatus_FINALITY_PROVIDER_STATUS_INACTIVE,
 			})
 		}
 
+		// Sort FPs exactly like the keeper does: first by voting power (descending),
+		// then by BTC public key hex (ascending) for equal voting power
+		sort.SliceStable(fpsWithMeta, func(i, j int) bool {
+			if fpsWithMeta[i].VotingPower == fpsWithMeta[j].VotingPower {
+				return fpsWithMeta[i].BtcPk.MarshalHex() < fpsWithMeta[j].BtcPk.MarshalHex()
+			}
+			return fpsWithMeta[i].VotingPower > fpsWithMeta[j].VotingPower
+		})
+
+		// Set expectations for the new simplified hooks
+		// Top numActiveFPs will enter the active set
+		for i := 0; i < numActiveFPs; i++ {
+			fHooks.EXPECT().AfterBbnFpEntersActiveSet(
+				gomock.Any(),
+				fpsWithMeta[i].Addr,
+			).Times(1)
+		}
+		// For inactive FPs, no hook calls are expected since they don't change active set status
+
+		// Set finalized epoch so FPs can have timestamped pub rand and become active
+		h.CheckpointingKeeperForFinality.EXPECT().GetLastFinalizedEpoch(gomock.Any()).Return(uint64(2)).AnyTimes()
+
 		// record voting power table
 		babylonHeight := datagen.RandomInt(r, 10) + 1
 		h.Ctx = datagen.WithCtxHeight(h.Ctx, babylonHeight)
 		h.BTCLightClientKeeper.EXPECT().GetTipInfo(gomock.Eq(h.Ctx)).Return(&btclctypes.BTCHeaderInfo{Height: 30})
 		h.BeginBlocker()
 
-		// assert that only top `min(MaxActiveFinalityProviders, numFPs)` finality providers have voting power
-		sort.SliceStable(fpsWithMeta, func(i, j int) bool {
-			return fpsWithMeta[i].VotingPower > fpsWithMeta[j].VotingPower
-		})
-		for i := 0; i < numActiveFPs; i++ {
-			votingPower := h.FinalityKeeper.GetVotingPower(h.Ctx, *fpsWithMeta[i].BtcPk, babylonHeight)
-			require.Equal(t, fpsWithMeta[i].VotingPower, votingPower)
-		}
-		for i := numActiveFPs; i < int(numFps); i++ {
-			votingPower := h.FinalityKeeper.GetVotingPower(h.Ctx, *fpsWithMeta[i].BtcPk, babylonHeight)
-			require.Zero(t, votingPower)
-		}
+		// for i := 0; i < numActiveFPs; i++ {
+		// 	votingPower := h.FinalityKeeper.GetVotingPower(h.Ctx, *fpsWithMeta[i].BtcPk, babylonHeight)
```

Comment:
Should we uncomment or remove this code?

Reply:
I had a goal to continue that testing at some point :sweat_smile:  

If possible I would like to leave it commented and continue to add testing in another PR